util/topsql/reporter/single_target.go

```diff
@@ -49,13 +52,37 @@ func NewSingleTargetDataSink() *SingleTargetDataSink {
 		curRPCAddr: "",
 		conn:       nil,
 		sendTaskCh: make(chan sendTask, 1),
+
+		registered: false,
+		registerer: registerer,
+	}
+
+	addr := config.GetGlobalConfig().TopSQL.ReceiverAddress
+	if addr != "" {
```

Comment:
If `addr` is empty, maybe no need to run the data sink at all?

Reply:
The design was initially designed to allow for address changes, and I respect that expectation here.